File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {

        int size = nums.size();

        if(size <= 2){
            return size;
        }
        int j = 2;

    

        for(int i=2; i<size; i++){
            if(nums[i] != nums[j-2]){
                swap(nums[i],nums[j]);
                j++;
            }
        }
        return j;
    }
};

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {

        int n = nums.length;
        if(n <= 2){
            return n;
        }

        int extraNum = 2;
        for(int i=2; i<n; i++){

            if(nums[i] != nums[extraNum - 2]){
                int temp = nums[i];
                nums[i] = nums[extraNum];
                nums[extraNum] = temp;

                extraNum++;
            }
            
        }
        return extraNum;
    }
}